### PR TITLE
Adding dependency injection with Koin

### PR DESCRIPTION
**Pull request by CorneilleEdi**
Adding **modules** for the DI and wrap all of them in the **AppComponent**.
Constructor injection in the viewmodel and injection of the viewModels in the fragments by using the `by viewModel()` DSL of Koin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ Also available in Play Store
 - [x] Offline Persistence with Room
 - [x] Databinding
 - [ ] ~[Jetpack Compose](https://github.com/mrcsxsiq/Kotlin-Pokedex/issues/4)~ - See [compose-pokedex](https://github.com/zsoltk/compose-pokedex)
-- [ ] Coroutines
+- [x] Coroutines
```
Comment (pdegand) [inline README.md]: Wrong line. You should check the Koin checkbox.

File: app/src/main/java/dev/marcosfarias/pokedex/App.kt [hunk 2]
```diff
@@ -4,26 +4,24 @@ import android.app.Application
...
-            AppDatabase::class.java,
-            getString(R.string.app_name)
-        )
-            .fallbackToDestructiveMigration()
-            .build()
+    }
 
+    private fun configureDI() = startKoin {
+        androidContext(this@App)
+        modules(provideComponent())
```
Comment (pdegand) [inline app/src/main/java/dev/marcosfarias/pokedex/App.kt]: why the function call ? Juste use `modules(appComponent)`

File: app/src/main/java/dev/marcosfarias/pokedex/di/NetworkModule.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+val networkModule = module {
+    single<Retrofit> {
+        Retrofit.Builder()
+            .baseUrl("https://gist.githubusercontent.com/mrcsxsiq/b94dbe9ab67147b642baa9109ce16e44/raw/97811a5df2df7304b5bc4fbb9ee018a0339b8a38/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
+
+    factory<PokemonService> {
+        get<Retrofit>().create(
```
Comment (pdegand) [inline app/src/main/java/dev/marcosfarias/pokedex/di/NetworkModule.kt]: use the Kotlin extension function for create: `get<Retrofit>().create()`

See https://github.com/square/retrofit/blob/master/retrofit/src/main/java/retrofit2/KotlinExtensions.kt#L29

File: app/src/main/java/dev/marcosfarias/pokedex/di/NetworkModule.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+val networkModule = module {
+    single<Retrofit> {
+        Retrofit.Builder()
+            .baseUrl("https://gist.githubusercontent.com/mrcsxsiq/b94dbe9ab67147b642baa9109ce16e44/raw/97811a5df2df7304b5bc4fbb9ee018a0339b8a38/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
+
+    factory<PokemonService> {
```
Comment (pdegand) [inline app/src/main/java/dev/marcosfarias/pokedex/di/NetworkModule.kt]: This should probably be a singleton as well.

File: app/build.gradle [hunk 5]
```diff
@@ -53,6 +53,14 @@ dependencies {
     kapt 'com.github.bumptech.glide:compiler:4.9.0'
     implementation 'com.github.bumptech.glide:glide:4.9.0'
 
+    // DI
+    // Koin for Android
+    implementation "org.koin:koin-android:2.0.1"
+    // or Koin for Lifecycle scoping
+    implementation "org.koin:koin-androidx-scope:2.0.1"
+    // or Koin for Android Architecture ViewModel
+    implementation "org.koin:koin-androidx-viewmodel:2.0.1"
```
Comment (pdegand) [inline app/build.gradle]: Follow the file convention and use single quotes.

**Comment by mrcsxsiq** [comment]
@CorneilleEdi @pdegand Thank you for your contributions

File: app/build.gradle (hunk 5 above)
Comment (mrcsxsiq) [reply app/build.gradle]: Indeed. A good next task in this project is to add ktlint. 
I'll try to do this in my free time.

https://ktlint.github.io/

File: app/src/main/java/dev/marcosfarias/pokedex/di/DatabaseModule.kt [hunk 6]
```diff
@@ -10,11 +10,11 @@ val databaseModule = module {
         Room.databaseBuilder(
             androidApplication(),
             AppDatabase::class.java,
-            "rapidsend.db"
+            getString(R.string.app_name)
```
Comment (mrcsxsiq) [inline app/src/main/java/dev/marcosfarias/pokedex/di/DatabaseModule.kt]: androidApplication().baseContext.getString(R.string.app_name)

**Pull request merged**